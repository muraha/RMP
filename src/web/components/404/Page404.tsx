import React from 'react'
import classnames from 'classnames'

import { Link } from 'react-router-dom'

import Button from '../utils/button/Button'
import Text, { TextSize, TextTransform, TextWeight } from '../utils/text/Text'

import styles from './styles.css'

const Page404: React.FC = () => {
  return (
    <>
      <div id={styles['pg404_background-404']}>
        <p className={styles.netflix}>
          <Text
            red
            size={TextSize.m}
            transform={TextTransform.lowercase}
            weight={TextWeight.extraBold}
          >
            netflix
          </Text>
          <Text
            red
            size={TextSize.m}
            transform={TextTransform.lowercase}
          >
            roulette
          </Text>
        </p>
        <div id={styles.pg404_stars1} />
        <div id={styles.pg404_stars2} />
        <div id={styles.pg404_stars3} />
        <div id={styles.pg404_title}>
          <span className={styles['pg404_text-small']}>
            THERE IS NO EMOTION, THERE IS
          </span>
          <br />
          <span className={styles['pg404_text-big']}>
            PEACE
          </span>
          <br />
          <span className={styles['pg404_text-small']}>
            THERE IS NO IGNORANCE, THERE IS
          </span>
          <br />
          <span className={styles['pg404_text-big']}>
            KNOWLEDGE
          </span>
          <br />
          <span className={styles['pg404_text-small']}>
            THERE IS NO PASSION, THERE IS
          </span>
          <br />
          <span className={styles['pg404_text-big']}>
            SERENITY
          </span>
          <br />
          <span className={styles['pg404_text-small']}>
            THERE IS NO CHAOS, THERE IS
          </span>
          <br />
          <span className={styles['pg404_text-big']}>
            HARMONY
          </span>
          <br />
          <span className={styles['pg404_text-small']}>
            THERE IS NO INFORMATION, THERE IS
          </span>
          <br />
          <span className={styles['pg404_text-404']}>
            404
          </span>
          <br />
          <br />
          <Link to="/"><Button gray classes={classnames(styles.button)}>GO BACK TO HOME</Button></Link>
        </div>
        <p className={styles.netflix_footer}>
          <Text
            red
            size={TextSize.m}
            transform={TextTransform.lowercase}
            weight={TextWeight.extraBold}
          >
            netflix
          </Text>
          <Text
            red
            size={TextSize.m}
            transform={TextTransform.lowercase}
          >
            roulette
          </Text>
        </p>
      </div>

    </>
  )
}

// eslint-disable-next-line import/no-default-export
export default Page404
