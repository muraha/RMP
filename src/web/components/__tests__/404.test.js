import React from 'react'
import toJson from 'enzyme-to-json'
import { cleanup } from '@testing-library/react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Page404 from '../404/Page404'

configure({ adapter: new Adapter() })

afterEach(cleanup)
describe('404 Page', () => {
  it('snapshot', () => {
    const wrapper = shallow(<Page404 />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
