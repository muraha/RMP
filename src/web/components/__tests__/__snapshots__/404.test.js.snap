// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 Page snapshot 1`] = `
<Fragment>
  <div
    id="pg404_background-404"
  >
    <p
      className="netflix"
    >
      <Text
        classes={null}
        red={true}
        size="m"
        transform="lowercase"
        weight="800"
      >
        netflix
      </Text>
      <Text
        classes={null}
        red={true}
        size="m"
        transform="lowercase"
        weight="400"
      >
        roulette
      </Text>
    </p>
    <div
      id="pg404_stars1"
    />
    <div
      id="pg404_stars2"
    />
    <div
      id="pg404_stars3"
    />
    <div
      id="pg404_title"
    >
      <span
        className="pg404_text-small"
      >
        THERE IS NO EMOTION, THERE IS
      </span>
      <br />
      <span
        className="pg404_text-big"
      >
        PEACE
      </span>
      <br />
      <span
        className="pg404_text-small"
      >
        THERE IS NO IGNORANCE, THERE IS
      </span>
      <br />
      <span
        className="pg404_text-big"
      >
        KNOWLEDGE
      </span>
      <br />
      <span
        className="pg404_text-small"
      >
        THERE IS NO PASSION, THERE IS
      </span>
      <br />
      <span
        className="pg404_text-big"
      >
        SERENITY
      </span>
      <br />
      <span
        className="pg404_text-small"
      >
        THERE IS NO CHAOS, THERE IS
      </span>
      <br />
      <span
        className="pg404_text-big"
      >
        HARMONY
      </span>
      <br />
      <span
        className="pg404_text-small"
      >
        THERE IS NO INFORMATION, THERE IS
      </span>
      <br />
      <span
        className="pg404_text-404"
      >
        404
      </span>
      <br />
      <br />
      <Link
        to="/"
      >
        <Button
          classes="button"
          gray={true}
        >
          GO BACK TO HOME
        </Button>
      </Link>
    </div>
    <p
      className="netflix_footer"
    >
      <Text
        classes={null}
        red={true}
        size="m"
        transform="lowercase"
        weight="800"
      >
        netflix
      </Text>
      <Text
        classes={null}
        red={true}
        size="m"
        transform="lowercase"
        weight="400"
      >
        roulette
      </Text>
    </p>
  </div>
</Fragment>
`;
